### PR TITLE
fixed #21

### LINKED ISSUE #21: Please separate Tests that use CrystCat
Some tests use functions from crystcat. As a result running the tests for cryst depends on crystcat which depends on cryst. This is bad for debian packaging, where as a result building gap-cryst requires gap-cryst to be installed.

Since I am packaging gap-cryst I have made a patch that separates the tests using crystcat and only runs them if crystcat can be loaded. You might want to apply it.
[tst-split.txt](https://github.com/gap-packages/cryst/files/5505992/tst-split.txt)

### PR DESCRIPTION
**Pull request by narc-Ontakac2**

### CODE REVIEW

File: tst/testall.g [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 
-TestDirectory( DirectoriesPackageLibrary( "cryst", "tst" ),
-  rec( exitGAP     := true,
+alltests := DirectoriesPackageLibrary( "cryst", "tst" );;
+crystcat := List(["crystcat.tst"], f -> Filename(alltests,f));;
+
+if LoadPackage( "crystcat" ) = true then
+  TestDirectory( crystcat,
+    rec( exitGAP     := false,
+         testOptions := rec( compareFunction := "uptowhitespace") ) );
```
Comment (fingolfin) [inline tst/testall.g:9]: This is not correct, as it ignores the result of the tests; the return value of `TestDirectory` would need to be honored.

But the better approach is to only call `TestDirectory` once, but change the exclusion list passed to it. Something like this (pseudo-code):
```gap
exclude := [];
if LoadPackage( "crystcat" ) = fail then
  exclude := ["crystcat.tst"];
fi;
TestDirectory( DirectoriesPackageLibrary( "cryst", "tst" ),
   rec( exclude := exclude, ...
```
Also, please don't use double semicolons needlessly.

**Review by fingolfin — CHANGES_REQUESTED** [review]
Thanks! But this will require some changes, I am afraid.

**Comment by narc-Ontakac2** [comment]
Done.

-- 
Papier ist gebundenes CO2. Bitte drucken Sie diese EMail aus und archivieren Sie sie.

**Review by fingolfin — APPROVED** [review]
LGTM, thanks! @gaehler OK to merge?

**Pull request merged**